crossbar/router/cookiestore.py

```diff
@@ -292,3 +295,19 @@ def setAuth(self, cbtid, authid, authrole, authmethod):
             if authid != cookie['authid'] or authrole != cookie['authrole'] or authmethod != cookie['authmethod']:
                 CookieStore.setAuth(self, cbtid, authid, authrole, authmethod)
                 self._persist(cbtid, cookie, status='modified')
+
+    def _clean_cookie_file(self):
+        with open(self._cookie_file_name, 'w') as cookie_file:
+            for cbtid, cookie in self._cookies.items():
+                cookie_record = json.dumps({
+                    'id': cbtid,
+                    'created': cookie['created'],
+                    'max_age': cookie['max_age'],
+                    'authid': cookie['authid'],
+                    'authrole': cookie['authrole'],
+                    'authmethod': cookie['authmethod']
+                }) + '\n'
+                cookie_file.write(cookie_record)
+
+            cookie_file.flush()
+            os.fsync(cookie_file.fileno())
```

Comment:
This probably isn't necessary; should flush on close() which will happen when the ``with`` block closes. But, probably no harm either...

Otherwise, LGTM +1